core/src/main/java/com/predic8/membrane/core/azure/api/auth/AuthenticationApi.java

```diff
@@ -49,10 +53,17 @@ public AuthenticationApi(HttpClient http, @Nullable AzureIdentity config) {
 
     public String accessToken() throws Exception {
         var response = http.call(tokenExchange()).getResponse();
-        return new ObjectMapper()
-                .readTree(response.getBodyAsStringDecoded())
-                .get("access_token")
-                .asText();
+        try {
+            return new ObjectMapper()
+                    .readTree(response.getBodyAsStringDecoded())
+                    .get("access_token")
+                    .asText();
+        } catch (Exception e) {
+            log.debug(e.getMessage());
+            log.debug(tokenExchange().getRequest().toString());
+            log.debug(tokenExchange().getResponse().toString());
```

Comment:
reading the body might not work, because `getBodyAsStringDecoded()` will already have read the body